### ISSUE TITLE
Reader.HasRows throws NullReferenceException when next result is not available

### ISSUE THREAD

**Issue by myshon** [snowflakedb/snowflake-connector-net#604]
### Issue description
reader.HasRows throws NullReferenceException when next result is not available. 

### Example code
Here is unit tests that reproduced the issue
```cs
    [Fact]
    public void ExecuteReader_WhenNoCommandLeft_HasRowsShouldReturnFalse()
    {
        using SnowflakeDbConnection conn = new SnowflakeDbConnection();
        conn.ConnectionString = "...";

        conn.Open();

        DbCommand cmd = conn.CreateCommand();
        var insertTemplate = "insert into db.public.output_table_01 select '{0}', '{1}', '{2}'";
        var cmdCommandText = string.Format(insertTemplate,
            Guid.NewGuid().ToString(),
            Guid.NewGuid().ToString(),
            Guid.NewGuid().ToString());
        
        var stmtCountParam = cmd.CreateParameter();
        stmtCountParam.ParameterName = "MULTI_STATEMENT_COUNT";
        stmtCountParam.DbType = DbType.Int16;
        stmtCountParam.Value = 3;
        cmd.Parameters.Add(stmtCountParam);
        
        cmd.CommandText = $"{cmdCommandText};{cmdCommandText};{cmdCommandText};";

        DbDataReader reader = cmd.ExecuteReader();
        
        Assert.False(reader.HasRows);
        Assert.Equal(1, reader.RecordsAffected);
        
        Assert.True(reader.NextResult());
        Assert.False(reader.HasRows);
        Assert.Equal(1, reader.RecordsAffected);
        
        Assert.True(reader.NextResult());
        Assert.False(reader.HasRows);
        Assert.Equal(1, reader.RecordsAffected);
        
        Assert.False(reader.NextResult());
        Assert.False(reader.HasRows); // --> throws NRE !!!
        
        reader.Close();
        
        conn.Close();
    }
```

### Error log
```
System.NullReferenceException
Object reference not set to an instance of an object.
   at Snowflake.Data.Core.ResultSetUtil.HasResultSet(SFBaseResultSet resultSet)
   at Snowflake.Data.Client.SnowflakeDbDataReader.get_HasRows()
```

### Configuration
*Driver version:* 2.0.20
*Dotnet framework and version:* NET6
*Server version:* 7.5.2
*Client OS:* Windows 10.

**Comment by sfc-gh-igarish** [comment]
Thanks for reporting issue. We are looking into it.

**Comment by sfc-gh-dszmolka** [comment]
fix is released with version 2.0.25, closing this issue
